src/lib/decorators.ts

```diff
@@ -265,3 +265,28 @@ export function eventOptions(options: AddEventListenerOptions) {
              // tslint:disable-next-line:no-any decorator
              any;
 }
+
+/**
+ * A property decorator that converts a class property into a getter that
+ * returns the `assignedNodes` of the given named `slot`.
+ *
+ * @ExportDecoratedItems
+ */
+export function queryAssignedNodes(slotName: string = '', flatten: boolean = false) {
+  return (protoOrDescriptor: Object|ClassElement,
+          // tslint:disable-next-line:no-any decorator
+          name?: PropertyKey): any => {
+    const descriptor = {
+      get(this: LitElement) {
+        const selector = `slot${slotName ? `[name=${slotName}]` : ''}`;
```

Comment:
Doesn't this need a selector before the `slot` tag name selector to work with ShadyDOM?

Reply:
The polyfilled `querySelector` works fine (it queries the logical tree).  It's only CSS selectors that require a wrapper, since they CSS only works on the actual tree.

Reply:
That's right, this works OK under polyfill. Note, it's only for styling that we need a node around slots when using ShadyDOM.